opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -62,6 +67,62 @@ def collect(self) -> Optional[_PointVarT]:
         pass
 
 
+class _AggregationFactory(ABC):
+    @abstractmethod
+    def _create_aggregation(self, instrument: Instrument) -> _Aggregation:
+        """Creates an aggregation"""
+
+
+class DefaultAggregation(_AggregationFactory):
+    """
+    The default aggregation to be used in a `View`.
+
+    This aggregation will create an actual aggregation depending on the
+    instrument type, as specified next:
+
+    `Counter`                   →   `SumAggregation`
+    `UpDownCounter`             →   `SumAggregation`
+    `ObservableCounter`         →   `SumAggregation`
+    `ObservableUpDownCounter`   →   `SumAggregation`
+    `Histogram`                 →   `ExplicitBucketHistogramAggregation`
+    `ObservableGauge`           →   `LastValueAggregation`
+    """
+
+    def _create_aggregation(self, instrument: Instrument) -> _Aggregation:
+
+        # pylint: disable=too-many-return-statements
+        if isinstance(instrument, Counter):
+            return _SumAggregation(
+                instrument_is_monotonic=True,
+                instrument_temporality=AggregationTemporality.DELTA,
+            )
+        if isinstance(instrument, UpDownCounter):
+            return _SumAggregation(
+                instrument_is_monotonic=False,
+                instrument_temporality=AggregationTemporality.DELTA,
+            )
+
+        if isinstance(instrument, ObservableCounter):
+            return _SumAggregation(
+                instrument_is_monotonic=True,
+                instrument_temporality=AggregationTemporality.CUMULATIVE,
+            )
+
+        if isinstance(instrument, ObservableUpDownCounter):
+            return _SumAggregation(
+                instrument_is_monotonic=False,
+                instrument_temporality=AggregationTemporality.CUMULATIVE,
+            )
+
+        if isinstance(instrument, Histogram):
+            return _ExplicitBucketHistogramAggregation()
+
+        if isinstance(instrument, ObservableGauge):
+            return _LastValueAggregation()
+
+        raise Exception("Invalid aggregation type found")
```

Comment:
You mean "invalid instrument type found"?

Reply:
Yes, fixed